superbench/runner/runner.py

```diff
@@ -58,6 +58,30 @@ def __get_enabled_benchmarks(self):
                 return list(self._sb_config.superbench.enable)
         return [k for k, v in self._sb_benchmarks.items() if v.enable]
 
+    def __get_mode_command(self, mode, exec_command):
+        """Get runner command for given mode.
+
+        Args:
+            mode (DictConfig): Runner mode.
+            exec_command (str): Executor command.
+
+        Return:
+            str: Runner command.
+        """
+        if mode.name == 'torch.distributed':
+            # TODO: replace with torch.distributed.run in v1.9
+            # TODO: only supports node_num=1 and node_num=all currently
+            return (
+                'python3 -m torch.distributed.launch '
+                '--use_env --no_python --nproc_per_node={proc_num} '
+                '--nnodes={node_num} --node_rank=$NODE_RANK '
+                '--master_addr=$MASTER_ADDR --master_port=$MASTER_PORT '
+                '{command}'
+            ).format(
+                proc_num=mode.proc_num or 8, node_num=1 if mode.node_num == 1 else '$NNODES', command=exec_command
```

Comment:
Who set $NODE_RANK, $NNODES, $MASTER_ADDR etc.? If node_num=1, master_addr should be `localhost`?

Reply:
it's set here:
https://github.com/microsoft/superbenchmark/blob/4063d8ebbd5760416f9e002514beccd070c5a483/superbench/runner/playbooks/check_env.yaml#L24-L28